src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,50 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private RegistrationServiceImp registrationService;
+    private final StorageDaoImpl storageMock;
```

Comment:
The `storageMock` should be of type `StorageDao` (the interface), not `StorageDaoImpl` (the implementation). This is because we want to program to an interface, not an implementation. This way, if we decide to change the implementation of `StorageDao` in the future, we won't need to change this line.